multi_camera_parallel_grabber/ex-multi_camera_parallel_grabber.cpp

```
@@ -0,0 +1,266 @@
+  const auto NUMBER_OF_DEVICES = 3;
+  const std::array<std::string, NUMBER_OF_DEVICES> camera_ips = { "192.168.0.70", "192.168.0.71","192.168.0.72"};
+
+  auto j_conf = json::parse(R"(
+        {
+          "ifm3d":
+          {
+            "Device":
+            {
+              "ActiveApplication": "1"
+            },
+            "Apps":
+            [
+              {
+                "Index": "1",
+                "TriggerMode":"2",
+                "LogicGraph": "{\"IOMap\": {\"OUT1\": \"RFT\",\"OUT2\": \"AQUFIN\"},\"blocks\": {\"B00001\": {\"pos\": {\"x\": 200,\"y\": 200},\"properties\": {},\"type\": \"PIN_EVENT_IMAGE_ACQUISITION_FINISHED\"},\"B00002\": {\"pos\": {\"x\": 200,\"y\": 75},\"properties\": {},\"type\": \"PIN_EVENT_READY_FOR_TRIGGER\"},\"B00003\": {\"pos\": {\"x\": 600,\"y\": 75},\"properties\": {\"pulse_duration\": 0},\"type\": \"DIGITAL_OUT1\"},\"B00005\": {\"pos\": {\"x\": 600,\"y\": 200},\"properties\": {\"pulse_duration\": 0},\"type\": \"DIGITAL_OUT2\"}},\"connectors\": {\"C00000\": {\"dst\": \"B00003\",\"dstEP\": 0,\"src\": \"B00002\",\"srcEP\": 0},\"C00001\": {\"dst\": \"B00005\",\"dstEP\": 0,\"src\": \"B00001\",\"srcEP\": 0}}}",
+                "Imager":
+                {
+                    "ExposureTime": "1000",
+                    "Type":"under5m_moderate",
+                    "FrameRate":"20"
+                }
+              }
+            ]
+          }
+        }
+    )");
+
+  //for configuration of required trigger on camera
+  void configuration(ifm3d::Camera::Ptr& camera, trigger_mode type)
+  {
+      try
+      {
+          auto application_id = camera->ActiveApplication();
+          j_conf["ifm3d"]["Device"]["ActiveApplication"] = std::to_string(application_id);
+          j_conf["ifm3d"]["Apps"][0]["Index"] = std::to_string(application_id);
+          j_conf["ifm3d"]["Apps"][0]["TriggerMode"] = std::to_string((int)type);
+
+          camera->FromJSON(j_conf);
+      }
+      catch (std::exception &e)
```

As a rule of thumb, catch const& not &.
Also, please catch ifm3d::error_t errors not global std::exception errors.

That said I suggest getting rid of the try/catch here and let the ctor throw on error. Otherwise the application will proceed and exhibit unexpected results.